src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs

```diff
@@ -272,14 +272,13 @@ public static SolutionStateChecksums Deserialize(ObjectReader reader)
 
                 if (projectState.TryGetStateChecksums(out var projectStateChecksums))
                 {
-                    if (searchingChecksumsLeft.Remove(projectStateChecksums.Checksum))
-                        result[projectStateChecksums.Checksum] = projectStateChecksums;
-
-                    if (searchingChecksumsLeft.Remove(projectStateChecksums.Info))
-                        result[projectStateChecksums.Info] = projectState.Attributes;
-
-                    if (searchingChecksumsLeft.Remove(projectStateChecksums.CompilationOptions))
-                        result[projectStateChecksums.CompilationOptions] = projectState.CompilationOptions!;
+                    await projectStateChecksums.FindAsync(
```

Comment:
this code was duplicating exactly what projectStateChecksums.FindAsync already does.  We just want to be able to search projects, just without a particular project-id provided. so this now defers to that helper to do that.  thanks the enum values this is easy.